programs/voter-stake-registry/src/instructions/extend_stake.rs

```diff
@@ -77,10 +77,12 @@ pub fn extend_stake(
         .checked_add(new_lockup_period.to_secs())
         .ok_or(VsrError::InvalidTimestampArguments)?;
     target.lockup.period = new_lockup_period;
+    target.delegate_mining = ctx.accounts.delegate_mining.key();
```

Comment:
On extending the stake we should not allow change of the delegate. For that we'll have a separate operation. I'd check for the equality `target.delegate_mining == ctx.accounts.delegate_mining.key()` instead

Reply:
Ok, will change. What's the reason for such a check? Delegate mining is not even a signer. It seems like, I had unified contexts for no reason.

Reply:
The reason is a simple sanity check that you know (as as client of the contract, like the FE dev) what you're doing. If you send a delegate A and the delegate B (who's on the deposit entry) is silently getting the stake - you'll not  be happy. On the other hand that means you'll need to provide delegate B in some other account for this to work, which doesn't seem feasible